marlowe-playground-client/src/Simulation.purs

```diff
@@ -118,11 +121,101 @@ initEditor editor =
         session <- Editor.getSession editor
         Session.setMode "ace/mode/haskell" session
 
-holesPane :: forall p. Array MarloweHole -> HTML p HAction
-holesPane holes = col3_ [ card_ [ cardBody_ (map displayHole holes) ] ]
+holesPane :: forall p. Maybe String -> Holes -> HTML p HAction
+holesPane selectedHole (Holes holes) =
+  let
+    kvs = Map.toUnfoldable holes
+
+    ordered = sortBy (\(Tuple _ a) (Tuple _ b) -> sortHoles a b) kvs
+
+    holesGroup = map (\(Tuple k v) -> displayHole selectedHole k v) ordered
+  in
+    col3_
+      [ div
+          [ class_ $ ClassName "btn-group-vertical"
+          , role "group"
+          ]
+          holesGroup
+      ]
+  where
+  sortHoles :: Array MarloweHole -> Array MarloweHole -> Ordering
+  sortHoles a b
+    | a == b = EQ
+    | otherwise =
+      fromMaybe EQ do
+        fa <- head a
+        fb <- head b
+        pure (compare fa fb)
```

Comment:
Could you just do `compare (head a) (head b)`?

Reply:
nice, I can actually change the whole thing to `compare `on` (head <<< snd)`